django/db/models/fields/__init__.py

```diff
@@ -144,6 +148,12 @@ def __init__(self, verbose_name=None, name=None, primary_key=False,
         self.remote_field = rel
         self.is_relation = self.remote_field is not None
         self.default = default
+        if db_default is not NOT_PROVIDED:
+            from django.db.models.expressions import Value
+            if not hasattr(db_default, 'resolve_expression'):
+                db_default = Value(db_default)
+            connection.ops.check_field_default_support(db_default)
```

Comment:
I'm not sure if this is the best place to raise an exception for invalid defaults.

It prevents using
```python
    class Meta:
        required_db_features = {'supports_functions_in_defaults'}
```
to conditionally define a model in the test suite. For example:

https://github.com/django/django/blob/d805f278123a0f6b768669c1f848cff75a4cec2b/tests/field_defaults/models.py#L50-L57

Reply:
@charettes How would you handle this?

Reply:
I guess we could defer `check_field_default_support` to a system check instead? It could also be performed in `db_default_sql` for good measure.